File: src/main/java/pl/arturczopek/mycoach/repository/ReportRepository.java
package pl.arturczopek.mycoach.repository;

import org.springframework.data.repository.PagingAndSortingRepository;
import pl.arturczopek.mycoach.model.database.Report;

import java.sql.Date;
import java.util.List;

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

public interface ReportRepository extends PagingAndSortingRepository<Report, Long> {

    @Override
    List<Report> findAll();

    List<Report> findAllByUserId(long userId);

    List<Report> findAllByUserIdOrderByEndDate(long userId);

    Report findFirstByEndDateBeforeAndReportIdNotOrderByEndDateDesc(Date date, long reportId);

    Report findFirstByStartDateAfterAndReportIdNotOrderByStartDate(Date date, long reportId);
}

File: src/main/kotlin/pl/arturczopek/mycoach/repository/ReportRepository.kt
package pl.arturczopek.mycoach.repository

import org.springframework.data.repository.PagingAndSortingRepository
import pl.arturczopek.mycoach.model.database.Report

import java.sql.Date

/**
 * @Author Artur Czopek
 * @Date 09-10-2016
 */

interface ReportRepository : PagingAndSortingRepository<Report, Long> {

    override fun findAll(): List<Report>

    fun findAllByUserId(userId: Long): List<Report>

    fun findAllByUserIdOrderByEndDate(userId: Long): List<Report>

    fun findFirstByEndDateBeforeAndReportIdNotOrderByEndDateDesc(date: Date, reportId: Long): Report

    fun findFirstByStartDateAfterAndReportIdNotOrderByStartDate(date: Date, reportId: Long): Report
}
